tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator/BuildTasks/GenerateFunctionsExtensionsMetadata.cs

```diff
@@ -44,31 +45,46 @@ public override bool Execute()
                 Arguments = $"Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator.Console.dll \"{SourcePath}\" \"{outputPath}\""
             };
 
-            var process = new Process { StartInfo = info };
-            process.EnableRaisingEvents = true;
-            process.ErrorDataReceived += (s, e) =>
+            Log.LogMessage(MessageImportance.Low, $"Extensions generator working directory: '{info.WorkingDirectory}'");
+            Log.LogMessage(MessageImportance.Low, $"Extensions generator path: '{info.FileName}'");
+            Log.LogCommandLine(MessageImportance.Low, info.Arguments);
+
+            using (var process = new Process { StartInfo = info })
             {
-                if (e.Data != null)
-                {
-                    Log.LogWarning(e.Data);
-                }
-            };
+                process.EnableRaisingEvents = true;
 
-            process.Start();
-            process.BeginErrorReadLine();
-            process.BeginOutputReadLine();
-            process.WaitForExit();
+                StringBuilder errorString = new StringBuilder();
+                process.ErrorDataReceived += (s, e) =>
+                {
+                    if (e.Data != null)
+                    {
+                        Log.LogWarning(e.Data);
+                        errorString.Append(e.Data);
+                    }
+                };
 
-            if (process.ExitCode != 0)
-            {
-                Log.LogError("Metadata generation failed.");
+                StringBuilder outputString = new StringBuilder();
+                process.OutputDataReceived += (s, e) =>
+                {
+                    if (e.Data != null)
+                    {
+                        outputString.Append(e.Data);
+                    }
+                };
 
-                return false;
-            }
+                process.Start();
+                process.BeginErrorReadLine();
+                process.BeginOutputReadLine();
+                process.WaitForExit();
 
-            process.Close();
+                if (process.ExitCode != 0)
```

Comment:
Nit: Do we need to execute Process.WaitForExit() here.

Reply:
We're doing it a couple of lines above. Or are you asking if we can remove that?